proto/cosmos/crisis/genesis.proto

```diff
@@ -6,11 +6,13 @@ option go_package = "github.com/cosmos/cosmos-sdk/x/crisis/types";
 import "gogoproto/gogo.proto";
 import "cosmos/cosmos.proto";
 
-// GenesisState - genesis state of x/crisis
+// GenesisState - genesis state of x/crisis.
 message GenesisState {
-	cosmos.Coin constant_fee = 3 [
-	  (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Coin",
-	  (gogoproto.nullable) = false,
-	  (gogoproto.moretags) = "yaml:\"constant_fee\""
-	];
+  // constant_fee is the fee used to verify the invariant in the crisis
+  // module.
+  cosmos.Coin constant_fee = 3 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Coin",
```

Comment:
We use both `casttype` and `customtype` in proto files, and I'm not sure what's the difference between the 2.

```
- customtype, It works with the Marshal and Unmarshal methods, to allow you to have your own types in your struct, but marshal to bytes. For example, custom.Uuid or custom.Fixed128
- casttype (beta), Changes the generated fieldtype.  All generated code assumes that this type is castable to the protocol buffer field type.  It does not work for structs or enums.
```

which one should we use here?

Reply:
I would propose to not use either but if we had to then the one not in beta would be better. When talking to the author of gogoproto he said these two features were a mistake to add.

Reply:
`customtype`.

Reply:
> When talking to the author of gogoproto he said these two features were a mistake to add.

Really, `customtype` is extremely helpful.

Reply:
Ah, so:
- `bytes address = 1 [(gogoproto.casttype) = "AccAddress"];` generates an `AccAddress`
- `  bytes address = 1 [(gogoproto.customtype) = "AccAddress"];` generates an `*AccAddress`

dunno... let's keep `casttype` then to avoid changing function signatures everywhere?

Reply:
You can just use `customtype` with nullable=false. But if this is regards to Genesis state, we should be using `string`.

Reply:
This annotation is actually unnecessary. `Coin` isn't a custom type it's proto generated.

```suggestion
```

Reply:
After switching to `customtype` everywhere, I ran into a dozen of failing tests. So let's move forward with this PR, and I/someone can revisit https://github.com/cosmos/cosmos-sdk/issues/7003.